src/CMakeLists.txt

```diff
@@ -412,6 +421,7 @@ set(common_SRCS
 	mapsector.cpp
 	metadata.cpp
 	modchannels.cpp
+	monitoring.cpp
```

Comment:
compile that only if USE_PROMETHEUS = true

Reply:
That won't work: there is a stub-implementation in there for the monitoring if the prometheus library is not found. It is either that or on every metric an `#if PROMETHEUS #endif` clause

Reply:
you should have instead an interface for this implementation and create a stub implementation when not enabled and a normal implementation when enabled.

Reply:
and also, replace monitoring -> metrology it's a better wording.

Reply:
inheritance: 
```
class IMetrology { // abstract }
class PrometheusMetrology : public IMetrology {}
class StubMetrology : public IMetrology {}
 ```

Reply:
I won't rename that to metrology, sounds too odd... :/